### PR TITLE
Sample app with the LuceneLinguistics

### PR DESCRIPTION
**Pull request by dainiusjocas**
I confirm that this contribution is made under the terms of the license found in the root directory of this repository's source tree and that I have the authority necessary to make this contribution on behalf of its copyright owner.

Given that Lucene linguistics has been already [released](https://github.com/vespa-engine/vespa/tree/master/lucene-linguistics), I believe it is worth adding several example applications.

Related open issue https://github.com/vespa-engine/vespa/issues/27912

NOTE: the min Vespa version that would support an optional `path` is 8.227.27.

### CODE REVIEW

**Comment by kkraune** [comment]
Discussed some changes with @dainiusjocas offline, waiting for a new iteration on this PR

**Comment by kkraune** [comment]
We are discussing in https://github.com/vespa-engine/vespa/issues/27912 whether this will work in multinode systems / if changes are needed - holding off this until we know more

**Comment by dainiusjocas** [comment]
Added an example on how to use LuceneLinguistics from non-java projects without pom.xml.

The bundle is hosts [here](https://github.com/dainiusjocas/vespa-lucene-linguistics-bundle/blob/main/README.md).

Or can I add the jar to this repository?

**Comment by kkraune** [comment]
I am adding @hmusum to this PR. He will work on the Path issue soon, so we can resolve that first to be able to simplify this

**Comment by dainiusjocas** [comment]
@hmusum [replied](https://github.com/vespa-engine/vespa/issues/27912) in the issue that a default value for the `path` type parameters is not going to be implemented.

@kkraune when can we come back to this PR?

**Comment by hmusum** [comment]
@dainiusjocas I am looking into supporting a `path` config type with optional value, I think that would solve your issues. It might take some time to implent, I'll keep you posted

**Comment by kkraune** [comment]
@hmusum we should look at this again now

**Comment by hmusum** [comment]
@dainiusjocas would you like to update this PR now that we have optional path support?

**Comment by dainiusjocas** [comment]
> @dainiusjocas would you like to update this PR now that we have optional path support?

@hmusum I've already updated the sample apps locally with `lucene-linguistics` built from source. I'm waiting for release [8.227.27](https://factory.vespa.oath.cloud/changes/8.227.27) that includes [optional path config](https://github.com/vespa-engine/vespa/pull/28529) or later to maven central (otherwise the code would not build).  The current version is [8.226.19](https://central.sonatype.com/search?q=lucene-linguistics)

**Comment by dainiusjocas** [comment]
@kkraune @hmusum, the sample apps are ready to review :)

File: examples/lucene-linguistics/minimal/README.md [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# Minimal `lucene-linguistics` setup
+
+This application package contains a bare minimal setup to get started with the `lucene-linguistics`.
+
+NOTE: the `ext/linguistics` directory can't be empty.
+The empty `ext/linguistics/dummy.txt` file serves no purpose except to make the Vespa configuration system happy ([details](https://github.com/vespa-engine/vespa/issues/27912)).
+When any other files are added to that directory then you could remove the `dummy.txt`.
```
Comment (hmusum) [inline examples/lucene-linguistics/minimal/README.md:7]: I guess these lines can be removed now that configDir is optional?

File: examples/lucene-linguistics/README.md [hunk 2]
```diff
@@ -0,0 +1,196 @@
+<!-- Copyright Yahoo. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root. -->
+
+![Vespa logo](https://vespa.ai/assets/vespa-logo-color.png)
+
+# Vespa LuceneLinguistics Demos
+
+A couple of example of how to get started with the `lucene-linguistics`:
+
+- `non-java`: an absolute minimum to get started; 
+  - TODO: update the bundle to the newest version when released
```
Comment (hmusum) [inline examples/lucene-linguistics/README.md:10]:
```suggestion

```

Remove?

**Review by hmusum — APPROVED** [review]
Two minor comments, otherwise looks good to me

File: examples/lucene-linguistics/minimal/README.md (hunk 1 above)
Comment (dainiusjocas) [reply examples/lucene-linguistics/minimal/README.md]: Yeah, they should be removed, thanks!

**Review by hmusum — APPROVED** [review]
👍

**Comment by kkraune** [comment]
This is a great contribution! This will help user of LuceneLinguistics a lot, and I will look at the rest of the Vespa docs / samples to integrate this where it makes sense.

**Pull request merged**